### DISCUSSION TITLE
is there any way to substitute the value and using contour.display to show the figure

### DISCUSSION THREAD

**Discussion by alwaysbyx** [ansys/pyfluent#3394] [Q&A] [answered]
Hi according to api reference,   
(1) we can use  `session.field_data.get_scalar_field_data` to get the raw value of the surface.    
(2) we can use `session.results.graphics.contour[object].display()` to display the contour image.

I wonder is there any way to substitute the value of that object's scalar data and display the resulted image

**Comment by prmukherj** [comment, marked as answer]
Hi @alwaysbyx,
If we understand your query correctly, you want to display your custom data in fluent interface using PyFluent. For this, we have a separate solution variables interface. Below is a sample script which can be used to achieve it:

>>> import ansys.fluent.core as pf
>>> s = pf.launch_fluent(ui_mode="gui")

>>> from ansys.fluent.core import examples
>>> import_file_name = examples.download_file("mixing_elbow.cas.h5", "pyfluent/mixing_elbow")
>>> examples.download_file("mixing_elbow.dat.h5", "pyfluent/mixing_elbow")

>>> solution_variable_data = s.fields.solution_variable_data

>>> s.file.read_case_data(file_name=import_file_name)

Create and display the original temperature data:
>>> s.results.graphics.contour["c1"] = {}
>>> c1 = s.results.graphics.contour["c1"]
>>> c1.field = "temperature"
>>> c1.surfaces_list = ["wall-elbow"]
>>> c1.display()

Create a dummy temperature data:
>>> wall_temp_array = solution_variable_data.create_empty_array("SV_T", "wall-elbow", "mixture")
>>> wall_temp_array[:] = 500

Update the data in Fluent:
>>> solution_variable_data.set_data(
>>> solution_variable_name="SV_T",
>>> zone_names_to_solution_variable_data={"wall-elbow": wall_temp_array},
>>> domain_name="mixture",
>>> )

Send a signal to Fluent that the data has been updated:
>>> s.scheme_eval.scheme_eval("(update-data-id)")

Again display the updated temperature data:
>>> c1.display()


We have a detailed documentation on using the solution-variable (https://fluent.docs.pyansys.com/version/stable/user_guide/fields/solution_data.html) services in PyFluent which you can refer for further information. An alternative to displaying in Fluent window is to use the pyfluent-visualization (https://github.com/ansys/pyfluent-visualization) as well.

Thanks @ajain-work for your advice.
cc. @seanpearsonuk 

Please let us know if this was helpful,
Thank you.

**Comment by prmukherj** [comment]
Hi @alwaysbyx,
We are converting this issue to discussion. Please let us know if the above solution works for you.
Thank you.
